src/app/beer_garden/db/mongo/models.py

```diff
@@ -924,14 +929,73 @@ class CommandPublishingBlocklist(Document):
 
 
 class Role(Document):
-    name = StringField()
+    name = StringField(required=True)
     description = StringField()
     permissions = ListField(field=StringField(), validation=validate_permissions)
 
     meta = {
         "indexes": [{"name": "unique_index", "fields": ["name"], "unique": True}],
     }
 
+    def __str__(self) -> str:
+        return self.name
+
+    def save(self, publish: bool = True, *args, **kwargs):
```

Comment:
I waffle back and forth on whether having the custom save here on the model is preferable, or if I should have mirrored what's done with the User model.  In that case, there is an `update_user` helper function that effectively wraps the save and publishes the event, rather than customizing the save method itself.  The difference there is that there's some special handling for the password field, though really that difference isn't so meaningful.

The benefit to having the custom save here is that you can just interact with the model naturally.  So the usage looks like:

```python
from beer_garden.db.mongo.models import Role
role = Role.objects.first()
role.description = "A truly great role"
role.save()
```

As opposed to for users:
```python
from beer_garden.db.mongo.models import User
from beer_garden.user import update_user
user = User.objects.first()
update_user(user, password="supersecret")
```

The User one requires the extra import and is also maybe less obvious that the save actually happened.

Maybe when in doubt go for consistency should have been the answer.  I figure I can solicit opinions before spending time changing it though.

Reply:
I"m sympathetic to the "interact with the model naturally" argument. *A foolish consistency is the hobgoblin of little minds...*